src/components/Input/Input.tsx

```diff
@@ -34,20 +35,24 @@ interface Input {
      * RegExp for input validation
      */
     pattern: RegExp;
+    togglePasswordVisability?: () => void;
 }
 
 const Input: FC<Input> = (props) => {
     const { t } = useTranslation();
     const errorMessage = (props.errors[props.name]?.message as string) || undefined;
     const id = useId();
+
+    const renderPasswordVisabilityIcon = () => (props.name === 'password' ? props.type === 'password' ? <HidePasswordIcon className={styles.input__hideIcon} onClick={props.togglePasswordVisability} /> : <ShowPasswordIcon className={styles.input__hideIcon} onClick={props.togglePasswordVisability} /> : null);
```

Comment:
```suggestion
    const visibilityIcon = (props.name === 'password' ? props.type === 'password' ? <HidePasswordIcon className={styles.input__hideIcon} onClick={props.togglePasswordVisability} /> : <ShowPasswordIcon className={styles.input__hideIcon} onClick={props.togglePasswordVisability} /> : null);
```
и там ниже
```
{visibilityIcon}
```
то есть, это можно сделать без функции, и без функции будет несколько проще, поэтому стоит делать без функции.

А вообще лучше сделать отдельный компонент PasswordInput который будет в себе это поведение прятать, и в месте его использования не надо будет заводить стейт isPasswordVisible